### PR TITLE
[wizard] create new import wizard project

### PR DESCRIPTION
**Pull request by chejennifer**
- initial commit to create a new React app for the import wizard
- added Mapping as a type and some function stubs so that frontend work can be done in parallel with heuristics and file generation

### CODE REVIEW

File: wizard/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "scripts": {
+    "build": "webpack --mode=production --progress --config webpack.config.js",
+    "start": "webpack serve --port 8080 --mode=development",
+    "predeploy": "npm run build",
+    "deploy": "gh-pages -d build",
+    "test": "jest"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (shifucun) [inline wizard/package.json:16]: what is ISC license?

File: wizard/src/utils/heuristics.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Mapping } from "../types";
+
+/**
+ * Given a csv, returns the predicted mapping
```
[the remark this answers is not in the document]
Comment (jehangiramjad) [reply wizard/src/utils/heuristics.ts:20]: Of course, type T in the output can also indicate that we failed to detect a type.

File: wizard/src/utils/heuristics.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * 
+ * @param csv array of csv rows where each row is an array of strings
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply wizard/src/utils/heuristics.ts:22]: yes, they are all parsed to strings

File: wizard/src/utils/heuristics.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (chejennifer) [reply wizard/src/utils/heuristics.ts:20]: if you look at the function definition, the return type is Mapping which I've defined here in [types.ts](https://github.com/datacommonsorg/import/pull/128/files/f598d00d260f7bc933abf176080d6e7350fd7709#diff-9081b6bfff71d505cc38e3eca03f17569cd88604a40ecd825f18dd23f486641aR46). I feel like this is kind of similar to what you're suggesting? I tried following the type for the translation metadata defined in the design doc

File: wizard/build/bundle.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/*! For license information please see bundle.js.LICENSE.txt */
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply wizard/build/bundle.js]: done, removed

File: wizard/package.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "wizard",
+  "version": "1.0.0",
+  "description": "",
+  "homepage": "https://datacommonsorg.github.io/import/",
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply wizard/package.json]: done

**Review by pradh — COMMENTED** [review]
very nice!

File: wizard/src/types.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  // Constant value as the mapping value. Should be set if type is
+  // MappingType.CONSTANT
+  constant?: string;
+}
+
+export interface Mapping {
+  place: MappingVal,
+  statVars: MappingVal,
+  dates: MappingVal,
+  units?: MappingVal
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply wizard/src/types.ts]: done

File: wizard/src/utils/file_generation.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+import { Mapping } from "../types";
+
+/**
+ * Generates the cleaned csv given the original csv and the correct mappings
+ * @param csv
+ * @param mappings
+ * @returns 
+ */
+export function generateCleanedCSV(csv: string[][], mappings: Mapping): string {
```
Comment (pradh) [inline wizard/src/utils/file_generation.ts:25]: CsvData?

**Review by pradh — APPROVED** [review]

File: wizard/src/utils/file_generation.ts (hunk 7 above)
Comment (pradh) [reply wizard/src/utils/file_generation.ts:25]: Ah, I was imagining CsvData will have the original csv (aka 2-d array), as well as any parsed stuff, so we have one object we pass around everywhere.

Comment (chejennifer) [reply wizard/src/utils/file_generation.ts]: ahh, ok, that makes sense, updated!

**Review by jehangiramjad — APPROVED** [review]
Thank you!

**Pull request merged**